src/cargo/util/network/http.rs

```diff
@@ -61,6 +61,16 @@ pub fn configure_http_handle(gctx: &GlobalContext, handle: &mut Easy) -> CargoRe
         let cainfo = cainfo.resolve_path(gctx);
         handle.cainfo(&cainfo)?;
     }
+    // Use `proxy_cainfo` if explicitly set; otherwise, fall back to `cainfo` as curl does #15376.
+    let effective_proxy_cainfo = match (&http.proxy_cainfo, &http.cainfo) {
+        (Some(p), _) => Some(p),
+        (None, Some(ca)) => Some(ca),
+        _ => None,
+    };
+    if let Some(proxy_cainfo) = effective_proxy_cainfo {
```

Comment:
This should be sufficient.

```suggestion
    if let Some(proxy_cainfo) = http.proxy_cainfo.as_ref().or(http.cainfo.as_ref()) {
```

Reply:
That is what I got on the first attempt, but wanted to make the fallback more visible. Accepted anyway.